// @vidstack/react/prod/player/vidstack-default-layout.js
"use client"

export { c as DefaultAudioLayout, h as DefaultBufferingIndicator, b as DefaultKeyboardDisplay, D as DefaultLayoutContext, l as DefaultMenuButton, j as DefaultMenuCheckbox, m as DefaultMenuItem, n as DefaultMenuRadioGroup, k as DefaultMenuSection, p as DefaultMenuSliderItem, q as DefaultSliderParts, r as DefaultSliderSteps, s as DefaultTooltip, g as DefaultVideoGestures, e as DefaultVideoLargeLayout, d as DefaultVideoLayout, f as DefaultVideoSmallLayout, o as createRadioOptions, i as i18n, u as useDefaultLayoutContext, a as useDefaultLayoutWord } from '../chunks/vidstack-Bb6CH4mU.js';
export { defaultLayoutIcons } from './vidstack-default-icons.js';
import 'react';
import '../chunks/vidstack-wClXxc1a.js';
import '../chunks/vidstack-DHQpOPJu.js';
import '../chunks/vidstack-3FvLzP0A.js';
import '../chunks/vidstack-mPfLbP-t.js';
import 'react-dom';
import '../chunks/vidstack-CBgf4xtl.js';
import '../chunks/vidstack-Dw1DfPZl.js';
import '../chunks/vidstack-CrXS53tu.js';
import '../chunks/vidstack-Xvq9wjeH.js';
